### DISCUSSION TITLE
[Adonis V5] How to manage nullable Date / DateTime in models?

### DISCUSSION THREAD

**Discussion by thespino** [adonisjs/core#1498] [Help] [answered]
Adonis version: **5.0.0-preview-rc-1.11**
Database: **mysql**

Hi, I've a model with a structure like like this:

```
class M extends BaseModel {
  ...

  @column.dateTime()
  nullableDateTime?: DateTime

  ...
}
```

In db, "nullableDateTime" is a datetime column which is nullable (and has default value of NULL). Now my question is: how can I edit this field when updating model to make it null again?

Let's talk with steps.

1) Creation
```
import { DateTime } from 'luxon'
...
await M.create({ nullableDateTime: DateTime.fromSeconds(...) })
```
And in database i have the correct datetime.

2) Update. I want to set datetime to NULL
```
import { DateTime } from 'luxon'
...
await M.update({ nullableDateTime: null })   // <-- This throws Typescript error "Type 'null' is not assignable to type 'DateTime | undefined'"
```
but if i make
```
import { DateTime } from 'luxon'
...
await M.update({ nullableDateTime: undefined })  // <-- This does NOT update value because it's not processed... it's undefined
```

Of course with a ts-ignore the error disappears and the first method works like a charm, but wondering if there's a better way of doing it and maintaining the types consistency.

Any ideas? Might this be an issue?

Thanks in advance

PS: the same happens with Date values

**Comment by thetutlage** [comment, marked as answer]
Yup. Types needs to be fixed here. If you can create a PR, then I merge and release it. Or at least create an issue with `@adonisjs/lucid` repo, so that it doesn't skip my mind

**Comment by thespino** [comment]
Created the issue https://github.com/adonisjs/lucid/issues/590

Thanks a lot. Marking as answered
